types/datum.go

```diff
@@ -1390,6 +1390,8 @@ func (d *Datum) ToBool(sc *stmtctx.StatementContext) (int64, error) {
 	case KindBinaryLiteral, KindMysqlBit:
 		val, err1 := d.GetBinaryLiteral().ToInt(sc)
 		isZero, err = val == 0, err1
+	case KindMysqlJSON:
+		isZero, err = d.IsNull(), nil
```

Comment:
It should be false if `$.test` is zero

Reply:
I had add testcase as follows:
a) testcase1:
insert into testjson set j='{"test":0}
select j from testjson where json_extract(j, '$.test')

b) testcase2:
insert into testjson set j='{"test1":0}
select j from testjson where json_extract(j, '$.test')

Reply:
I've tested this case in Mysql5.7, and the result is different from yours.
```mysql
mysql> drop table testjson;
Query OK, 0 rows affected (0.00 sec)

mysql> create table testjson(j json );
Query OK, 0 rows affected (0.01 sec)

mysql> insert into testjson set j='{"test":0}';
Query OK, 1 row affected (0.00 sec)

mysql> select j from testjson where json_extract(j, '$.test');
Empty set (0.00 sec)
```

Reply:
Another case is different from Mysql5.7

In your branch:
```mysql
mysql> create table testjson(j json);
Query OK, 0 rows affected (0.01 sec)

mysql> insert into testjson set j='0';
Query OK, 1 row affected (0.00 sec)

mysql> insert into testjson set j='null';
Query OK, 1 row affected (0.00 sec)

mysql> insert into testjson set j='1';
Query OK, 1 row affected (0.00 sec)

mysql> select j from testjson where j;
+------+
| j    |
+------+
| 0    |
| null |
| 1    |
+------+
3 rows in set (0.00 sec)
```

In MySQL5.7
```mysql
mysql> create table testjson(j json);
Query OK, 0 rows affected (0.02 sec)

mysql> insert into testjson set j='0';
Query OK, 1 row affected (0.01 sec)

mysql> insert into testjson set j='null';
Query OK, 1 row affected (0.00 sec)

mysql> insert into testjson set j='1';
Query OK, 1 row affected (0.00 sec)

mysql> select j from testjson where j;
+------+
| j    |
+------+
| 1    |
+------+
1 row in set, 1 warning (0.00 sec)

mysql> show warnings;
+---------+------+---------------------------------------------------------------+
| Level   | Code | Message                                                       |
+---------+------+---------------------------------------------------------------+
| Warning | 3156 | Invalid JSON value for CAST to INTEGER from column j at row 2 |
+---------+------+---------------------------------------------------------------+
1 row in set (0.00 sec)
```

Reply:
Thank you, I will investigate it.